sdk/examples/wallet/wallet.rs

```diff
@@ -1,106 +1,168 @@
 // Copyright 2021 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-//! TODO: Example description
+//! In this example we will:
+//! * restore a wallet from a mnemonic phrase
+//! * create an account if it does not exist yet
+//! * generate some addresses for that account - if necessary
+//! * print all addresses in the account
+//! * print all addresses with funds in the account
+//! * make a coin transaction
 //!
-//! `cargo run --example wallet --release`
+//! Make sure there's no `example.stronghold` file and no `example.walletdb` folder yet!
+//!
+//! Rename `.env.example` to `.env` first, then run the command:
+//! ```sh
+//! cargo run --release --all-features --example wallet
+//! ```
 
-use std::time::Instant;
+use std::{env::var, time::Instant};
 
 use iota_sdk::{
     client::{
         constants::SHIMMER_COIN_TYPE,
         secret::{mnemonic::MnemonicSecretManager, SecretManager},
     },
-    wallet::{ClientOptions, Result, SendAmountParams, Wallet},
+    types::block::payload::transaction::TransactionId,
+    wallet::{Account, ClientOptions, Result, SendAmountParams, Wallet},
 };
 
+// The number of addresses to generate in this account
+const MAX_ADDRESSES_TO_GENERATE: usize = 10;
+// The amount of coins to send
+const SEND_AMOUNT: u64 = 1_000_000;
+// The address to send the coins to
+const RECV_ADDRESS: &str = "rms1qpszqzadsym6wpppd6z037dvlejmjuke7s24hm95s9fg9vpua7vluaw60xu";
+
 #[tokio::main]
 async fn main() -> Result<()> {
     // This example uses secrets in environment variables for simplicity which should not be done in production.
     dotenvy::dotenv().ok();
 
-    let client_options = ClientOptions::new().with_node(&std::env::var("NODE_URL").unwrap())?;
+    let wallet = restore_wallet().await?;
 
-    let secret_manager =
-        MnemonicSecretManager::try_from_mnemonic(&std::env::var("NON_SECURE_USE_OF_DEVELOPMENT_MNEMONIC_1").unwrap())?;
+    let alias = var("ACCOUNT_ALIAS_1").unwrap();
+    let account = get_or_create_account(&wallet, &alias).await?;
+    print_accounts(&wallet).await?;
+
+    generate_addresses(&account, MAX_ADDRESSES_TO_GENERATE).await?;
+    print_addresses(&account).await?;
+
+    // Change to `true` to print the full balance report
+    sync_print_balance(&account, false).await?;
 
-    let wallet = Wallet::builder()
+    print_addresses_with_funds(&account).await?;
+
+    println!("Sending '{}' coins to '{}'...", SEND_AMOUNT, RECV_ADDRESS);
+    let outputs = [SendAmountParams::new(RECV_ADDRESS, SEND_AMOUNT)?];
+    let transaction = account.send_amount(outputs, None).await?;
+    wait_for_inclusion(&transaction.transaction_id, &account).await?;
+
+    sync_print_balance(&account, false).await?;
+
+    println!("Example finished successfully");
+    Ok(())
+}
+
+async fn restore_wallet() -> Result<Wallet> {
```

Comment:
Related to the comment above, don't see why this is restoring

Reply:
Changed